crates/policy-evaluator/src/cluster_context.rs

```diff
@@ -0,0 +1,96 @@
+#![allow(clippy::mutex_atomic)]
+
+use anyhow::Result;
+use kube::api::{ListParams, Resource};
+use kube::Client;
+use std::sync::{Mutex, RwLock};
+use tokio::runtime::Builder;
+
+use k8s_openapi::api::core::v1::{Namespace, Service};
+use k8s_openapi::api::networking::v1::Ingress;
+
+use lazy_static::lazy_static;
+use std::thread;
+
+lazy_static! {
+    static ref CLUSTER_CONTEXT_INITIALIZED: Mutex<bool> = Mutex::new(false);
+    static ref CLUSTER_CONTEXT: ClusterContext = ClusterContext::default();
+}
+
+#[derive(Default)]
+pub struct ClusterContext {
```

Comment:
Please add some inline documentation explaining:
  * What is the ClusterContext struct, what's its purpose
  * How the synchronization works: all the part about RW locks
  * What are the limitations, plans for the future. Basically what you already wrote inside of the PR body